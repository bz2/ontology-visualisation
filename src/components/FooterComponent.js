import React from 'react'
import { connect } from 'redux-zero/react'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import actions from '../store/actions'

const FooterComponent = ({
  availableEdgesNormalised,
}) => {
  const { t } = useTranslation()

  return (
    <footer>
      <div className="footer-left">
        <span>
          {`${t('edges')}: ${Object.keys(availableEdgesNormalised).length}`}
        </span>
      </div>
    </footer>
  )
}

FooterComponent.propTypes = {
  availableEdgesNormalised: PropTypes.shape().isRequired,
}

const mapToProps = ({
  availableEdgesNormalised,
}) => ({
  availableEdgesNormalised,
})

export default connect(
  mapToProps,
  actions
)(FooterComponent)
