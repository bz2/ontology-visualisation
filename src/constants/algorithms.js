export const ALGO_TYPE_FULL = 'full'
export const ALGO_TYPE_NEIGHBOURHOOD = 'neighbourhood'
export const ALGO_TYPE_SHORTEST_PATH = 'shortest-path'
export const ALGO_TYPE_BOUNDING_BOX = 'bounding-box'
export const ALGO_TYPE_NODES_FILTER = 'nodes-filter'
export const ALGO_TYPE_EDGES_FILTER = 'edges-filter'
