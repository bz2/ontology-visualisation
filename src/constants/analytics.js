export const USERBACK_ACCESS_TOKEN = '29230|42198|CVJlhtfkaYN3J3xP2x200ghJ1'
export const USERBACK_COOKIE = 'ubwc'
export const USERBACK_WIDGET = 'https://static.userback.io/widget/v1.js'
export const GA_ID = 'UA-195025859-1'
