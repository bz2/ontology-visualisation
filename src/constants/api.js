export const API_ENDPOINT = `${process.env.endpointUrl}/api` // 'http://localhost:8080/api'

export const GET_ONTOLOGY = `${API_ENDPOINT}/ontology`
export const GET_GRAPH = `${API_ENDPOINT}/graph`
export const GET_GRAPH_QUERY = `${GET_GRAPH}/query`

export const GET_EDGES = `${GET_GRAPH}/edges`
export const GET_EDGE = `${GET_GRAPH}/edges/{id}`

export const GET_NODES = `${GET_GRAPH}/nodes`
export const GET_NODE = `${GET_GRAPH}/nodes/{id}`

// export const GET_PROPERTIES = `${GET_GRAPH}/properties`
// export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
// export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
// export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
// export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`
export const GET_PROPERTIES = `${GET_ONTOLOGY}/properties`
export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`

export const AUTH_SIGN_IN = `${API_ENDPOINT}/login`
