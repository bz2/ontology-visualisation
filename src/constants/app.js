export const APP_NAME = 'OntoPop'
