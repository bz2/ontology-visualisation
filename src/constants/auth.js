export const AUTH_COOKIE = 'ontology_viz_auth'
