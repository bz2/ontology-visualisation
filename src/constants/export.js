export const EXPORT_JPG = 'jpeg'
export const EXPORT_PNG = 'png'
export const EXPORT_PDF = 'pdf'
export const EXPORT_CSV = 'csv'
export const EXPORT_OWL = 'owl'

export const EXPORT_GRAPH_OPTIONS = [
  EXPORT_JPG,
  EXPORT_PNG,
  EXPORT_PDF
]

export const EXPORT_DATA_OPTIONS = [
  EXPORT_CSV,
  EXPORT_OWL
]
