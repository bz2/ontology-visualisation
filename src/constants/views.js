export const SIDEBAR_VIEW_GRAPHS = 'networkGraphs'
export const SIDEBAR_VIEW_FREE_TEXT_SEARCH = 'freeTextSearch'
export const SIDEBAR_VIEW_NODES_SELECTION = 'nodesSelection'
export const SIDEBAR_VIEW_EDGES_SELECTION = 'edgesSelection'
export const SIDEBAR_VIEW_BOUNDING_BOX = 'boundingBox'
export const SIDEBAR_VIEW_NEIGHBOURHOOD = 'nodeNeighbourhood'
export const SIDEBAR_VIEW_SHORTEST_PATH = 'shortestPath'
export const SIDEBAR_VIEW_SETTINGS = 'settings'
export const SIDEBAR_VIEW_EXPORT = 'export'
export const SIDEBAR_VIEW_CUSTOM_QUERY = 'customQuery'
