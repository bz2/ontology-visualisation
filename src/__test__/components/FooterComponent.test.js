import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FooterComponent from '../../components/FooterComponent'
import { SIDEBAR_VIEW_ENTRY_SEARCH } from '../../constants/views'

const setup = ({
  sidebarView,
  totalSearchCount
}) => {
  const props = {
    updateStoreValue: jest.fn(),
    totalSearchCount,
    sidebarView,
    searchPageSelected: 1,
    entrySearchResultsByPage: {
      0: [{
        id: '1'
      }],
      1: [{
        id: '2'
      }],
    }
  }

  const component = shallow(<FooterComponent {...props} />)

  return {
    component,
    props
  }
}

describe('FooterComponent', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot when SIDEBAR_VIEW_ENTRY_SEARCH and 0 total search count', () => {
    const {
      component
    } = setup({
      sidebarView: SIDEBAR_VIEW_ENTRY_SEARCH,
      totalSearchCount: 0,
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when SIDEBAR_VIEW_ENTRY_SEARCH', () => {
    const {
      component
    } = setup({
      sidebarView: SIDEBAR_VIEW_ENTRY_SEARCH,
      totalSearchCount: 100,
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when any  view', () => {
    const {
      component
    } = setup({
      sidebarView: 'any',
      totalSearchCount: 0,
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
