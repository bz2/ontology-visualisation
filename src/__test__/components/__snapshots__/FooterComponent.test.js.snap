// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should match snapshot  1`] = `
<footer>
  <div
    className="footer-right"
  >
    Ontology Visualisation v0.6.0
  </div>
</footer>
`;
